### PR TITLE
CATROID-873 Pocket Code Beta flavor

### PR DESCRIPTION
**Pull request by dandrawes**
https://jira.catrob.at/browse/CATROID-873
- add Pocket Code BETA flavor
- revert unneeded changes from: https://github.com/Catrobat/Catroid/commit/30640a77e6a41816d083ac4994ae51ac026495f3
- changed version text in About Dialog

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by juliajulie95 — CHANGES_REQUESTED** [review]
jenkins build with 'BUILD_ALL_FLAVORS' parameter fails
debug label not set in serializer

**Review by juliajulie95 — APPROVED** [review]

File: fastlane/Fastfile [hunk 1]
```diff
@@ -257,6 +257,53 @@ platform :android do
...
+        json_key_data:              ENV['googlePlayJsonKey'],
+        metadata_path:              './fastlane/metadata/android',
+        skip_upload_apk:            'true',
+        skip_upload_metadata:       'true',
+        skip_upload_images:         'true',
+        skip_upload_screenshots:    'true',
+      )
+    end
+
   desc "upload Playground metadata only"
```
Comment (wslany) [inline fastlane/Fastfile:307]: No playground version anymore.

File: fastlane/Fastfile [hunk 2]
```diff
@@ -257,6 +257,53 @@ platform :android do
...
+        track:                      'alpha',
+        track_promote_to:           'production',
+        json_key_data:              ENV['googlePlayJsonKey'],
+        metadata_path:              './fastlane/metadata/android',
+        skip_upload_apk:            'true',
+        skip_upload_metadata:       'true',
+        skip_upload_images:         'true',
+        skip_upload_screenshots:    'true',
+      )
+    end
```
Comment (wslany) [inline fastlane/Fastfile:305]:
```suggestion
```
Actually, the beta version shall _never_ be promoted to production (because we do not want visible reviews for it, among other reasons), so this part can be deleted.

**Review by wslany — CHANGES_REQUESTED** [review]
See inlined comments.

One more thing: the versionName in https://github.com/Catrobat/Catroid/blob/7399877a67a8980a9305a7f84bb76f62398df79f/catroid/build.gradle#L130 shall be appended with " BETA-X" where X is the versionCode defined in line [129](https://github.com/Catrobat/Catroid/blob/7399877a67a8980a9305a7f84bb76f62398df79f/catroid/build.gradle#L129), i.e., for instance "0.9.75 BETA-80".

**Review by mstoeg — APPROVED** [review]
Well done!

**Review by wslany — CHANGES_REQUESTED** [review]
The app version name in Pocket Code incorrectly is shown as "null":
image

Same also in the beta version:
image

**Comment by wslany** [comment]
To make sure this cannot easily happen again, since it can apparently go wrong without us noticing it immediately, please include a test that checks that the app version is not null or empty.

**Review by wslany — APPROVED** [review]

**Comment by wslany** [comment]
Thanks!
